### PR TITLE
feat(lql): add limit option 

### PR DESCRIPTION
**Pull request by hazedav**
## Summary
As a Lacework user, I want to be able to limit LQL query results

`lacework query run LW_Global_AWS_CTA_S3BucketPolicyChange --limit 10`

## How did you test this change?
Manually
Automated Unit + Integration

## Issue
https://lacework.atlassian.net/browse/ALLY-1102

### CODE REVIEW

File: api/lql_execute.go [hunk 1]
```diff
@@ -38,18 +38,34 @@ const (
 	QueryEndTimeRange   ExecuteQueryArgumentName = "EndTimeRange"
 )
 
+type ExecuteQueryOptions struct {
+	Limit *int `json:"limit,omitempty"`
+}
```
Comment (afiune) [inline api/lql_execute.go:43]: Are there any more query options?

File: cli/cmd/lql.go [hunk 2]
```diff
@@ -153,34 +159,41 @@ func init() {
 	// run specific flags
 	setQuerySourceFlags(queryRunCmd)
 
-	// since time flag
-	queryRunCmd.Flags().StringVarP(
+	// limit flag
+	queryRunCmd.Flags().IntVar(
+		&queryCmdState.Limit,
+		"limit", 0,
```
Comment (afiune) [inline cli/cmd/lql.go:165]: Is this the only command we should add this new flag?

File: api/lql_execute.go (hunk 1 above)
Comment (hazedav) [reply api/lql_execute.go:43]: This is the only one.

File: cli/cmd/lql.go (hunk 2 above)
Comment (hazedav) [reply cli/cmd/lql.go:165]: Yes, my understanding of this flag (similar to the parameters) is that `limit` is only applicable to query execution

Comment (hazedav) [reply cli/cmd/lql.go:164]: I double checked the API docs and confirmed this.

File: api/lql_execute.go [hunk 3]
```diff
@@ -38,18 +38,34 @@ const (
...
+func (eqo ExecuteQueryOptions) Validate() error {
+	if eqo.Limit == nil {
+		return nil
+	}
+	if *eqo.Limit >= 1 {
+		return nil
+	}
+	return errors.New("limit must be at least 1")
+}
+
```
Comment (afiune) [inline api/lql_execute.go:54]: Do we need to implement all of that? Shouldn't this be enforced by the APIs?

Comment (afiune) [reply api/lql_execute.go:54]: Testing passing a negative value does show:
```
[400] Incorrect configuration of field options.limit: must be greater than or equal to 1
```

Comment (afiune) [reply api/lql_execute.go:54]: Up to you to leave it, this is just a nitpick since you already implemented the logic/code.

**Review by afiune — APPROVED** [review]
## That was fast!
tenor-134928607

File: api/lql_execute.go (hunk 3 above)
Comment (hazedav) [reply api/lql_execute.go:54]: I worked this up based on some early go-sdk feedback you had given about reducing API calls if we're able to validate client side.  The pro is the ability to reduce server-side calls, but the con is having to keep in-line with platform validation.

If you're good with it I might chose to remove this in favor of server-side validation...

David

**Pull request merged**